fmriprep/workflows/bold/confounds.py

```diff
@@ -270,6 +270,55 @@ def _pick_wm(files):
     return workflow
 
 
+def init_carpetplot_wf(mem_gb, metadata, name="bold_confs_wf"):
+
+    inputnode = pe.Node(niu.IdentityInterface(
+        fields=['bold', 'bold_mask', 'confounds_file', 'bold_mni_transforms']),
+        name='inputnode')
+
+    # Warp segmentation into EPI space
+    resample_parc = pe.Node(ApplyTransforms(
+        float=True,
+        input_image=os.path.join(
+            get_mni_icbm152_nlin_asym_09c(), '1mm_parc.nii.gz'),
+        dimension=3, default_value=0, interpolation='MultiLabel'),
+        name='resample_parc')
+
+    # Carpetplot and confounds plot
+    conf_plot = pe.Node(FMRISummary(
+        tr=metadata['RepetitionTime'],
+        confounds_list=['GlobalSignal', 'CSF', 'WhiteMatter', 'stdDVARS',
+                        ('FramewiseDisplacement', 'mm')]),
+        name='conf_plot', mem_gb=mem_gb)
+    ds_report_bold_conf = pe.Node(
+        DerivativesDataSink(suffix='carpetplot'),
+        name='ds_report_bold_conf', run_without_submitting=True,
+        mem_gb=DEFAULT_MEMORY_MIN_GB)
+
+    def _reversed(inlist):
+        return list(reversed(inlist))
+
+    def _flags(inlist):
+        return [True] * len(inlist)
+
+    workflow = pe.Workflow(name=name)
+    workflow.connect([
+        (inputnode, resample_parc, [
+            ('bold_mask', 'reference_image'),
+            (('bold_mni_transforms', _reversed), 'transforms'),
```

Comment:
Is this going to be correct, given that the ANTs warps aren't invertible? We actually need `t1_2_mni_reversed_transform`.

Reply:
Okay, now I recall why I did it this way.

In principle, you are right: we should use the `t1_2_mni_reversed_transform` and the reversed transform of the SDC.

However, I preferred to set the `invert_transform_flags` and have ApplyTransform to calculate the inverses. You are right that some transforms may not be invertible (although SyN is diffeomorphic so the T1-to-MNI should be invertible), but ApplyTransform will approximate the transforms and this way was very convenient. Otherwise, we would need to keep track of which transforms are inverted. Also, I'm not sure whether we generate the warped-to-corrected transform for SDC. If not, then we will need to approximate anyways (and it will be likely not invertible for EPI and fieldmap SDCs).

So far, with the examples I've seen, the estimated transforms seem accurate enough to resample the atlas in bold space for the purpose of the carpet plot.

Sounds reasonable?

Reply:
Sure. Approximate seems fine, here.